.github/actions/ci-setup/action.yml

```diff
@@ -0,0 +1,15 @@
+name: CI setup
```

Comment:
Using composite action I've extracted the common setup+install steps into a single block. You already had some small differences in those steps within your workflows.

Reply:
What's the point of reusing a few lines of code? You don't even need some of those lines. This way log will be good enough

```
    - uses: actions/setup-node@v3
      with:
        node-version: 16
        cache: "yarn"
    - run: yarn install --frozen-lockfile --ignore-scripts
```

Reply:
I don't feel super strong about this. It's a strategy I have used in the past in some other repos and I just used it here as well, since you already had different actions doing this thing slightly differently. So to keep things consistent, I've moved this to shared action.